contributors/design-proposals/local-ephemeral-storage.md

```diff
@@ -0,0 +1,399 @@
+
+# Design Overview
+This feature adds only one storage API to represent ephemeral storage from root partition and manage its isolation. The management of local ephemeral storage is consistent with memory management.
+```
+// Local ephemeral storage for root partition
+ResourceEphemeralStorage ResourceName = "ephemeral-storage"
+```
+
+## Resource Request/Limit
+* Container-level resource requirement: Similar to CPU and memory, container spec can specify the request/limit for local ephemeral storage. All the validation related to the storage resource requirements will be the same as memory resource. As mentioned above in  Storage types and configuration, kubernetes support optional imagefs partition. In the case of two partition, only the storage usage in root partition will be managed and isolated across different containers. (In another word, container writable layer usage will not be counted for container-level eviction management)
+* Pod-level resource constraint: Since we haven’t supported pod-level resource API, the sum of the resource request/limit is considered as pod-level resource requirement. Similar to memory, all local ephemeral storage resource usage is subject to this requirement. For example, emptyDir disk usage (also secrets, configuMap, downwardAPI, gitRepo since they are wrapped emptyDir volume) plus containers disk usage should not exceed the the pod-level resource requirement (the sum of all container’s limit).
+* EmptyDir SizeLimit: Because empryDir volume is a pod-level resource which is managed separately, we also add a sizeLimit for emptyDir Volume for additional storage isolation. If this limit is set for emptyDir volume (default medium), eviction manager will validate this limit with emptyDir usage too in additional to the above pod-level resource constraints. 
+
+## Eviction Policy and Scheduler Predicates
+CPU and memory use cgroup for limiting the resource usage. However, cgroup for disk usage is not available. Our current design is to evict pods when exceeding the limit set for local storage. The eviction policy is listed as follows.
+* If the container writable layer (overlay) usage exceeds its limit, pod gets evicted.
+* If the emptyDir volume usage exceeds the sizeLimit, pod gets evicted.
+* If the sum of the usage from emptyDir and all contains exceeds the sum of the container’s local storage limits (pod-level limit), pod gets evicted.
+
+When scheduler admits pods, it sums up the storage requests from each container and pod can be scheduled only if the sum is smaller than the allocatable of the local storage space.
+
+## Resource Quota
+Add two more resource quotas for storage. The request and limit set constraints on the total requests/limits of all containers’ in a namespace 
+ResourceRequestsStorageScratch
```

Comment:
we changed from scratch to ephemeral?
s/ResourceRequestsStorageScratch/ResourceRequestsEphemeralStorage
s/ResourceLimitStorageScratch/ResourceLimitsEphemeralStorage

Also, adding newline in markdown won't render new lines, you can change to use bullet list.

Reply:
fixed